src/nvim/api/options.c

```diff
@@ -343,6 +343,7 @@ Dictionary nvim_get_option_info2(String name, Dict(option) *opts, Error *err)
 
 /// Sets the global value of an option.
 ///
+/// @deprecated
```

Comment:
ideally these should move to `api/deprecated.c`

Reply:
> Is `access_option_value_for` needed for anything else? If not then move to deprecated.c. Ideally we should eliminate these.

Yes, it's needed for `nvim_get_option_value`

Reply:
> Which deprecated API is it used by? I can only see it being used by new APIs.

It's used by `set_option_to` which itself is deprecated since it's only used by other deprecated functions